draft-ietf-tls-tls13.md

```diff
@@ -4128,9 +4212,174 @@ Section 7 of [RFC6066], as it is not applicable to AEAD ciphers and has
+then the output keys are forward secret.
+
+The PSK and resumption-PSK modes bootstrap from a long-term shared
+secret into a unique per-connection short-term session key. This
+secret may have been established in a previous handshake. If
+PSK-(EC)DHE modes are used, this session key will also be forward
+secret. The resumption-PSK mode has been designed so that the
+resumption master secret computed by connection N and needed to form
+connection N+1 is separate from the traffic keys used by connection N,
+thus providing forward secrecy between the connections.
+
+For all handshake modes, the Finished MAC (and where present, the
+signature), prevents downgrade attacks. In addition, the use of
+certain bytes in the random nonces as described in {{server-hello}}
+allows the detection of downgrade to previous TLS versions.
+
+As soon as the client and the server have exchanged enough information
+to establish shared keys, the remainder of the handshake is encrypted,
+thus providing protection against passive attackers. Because the server
+authenticates before the client, the client can ensure that it only
+reveals its identity to an authenticated server. Note that implementations
+must use the provided record padding mechanism during the handshake
+to avoid leaking information about the identities due to length.
+
+The 0-RTT mode of operation generally provides the same security
+properties as 1-RTT data, with the two exceptions that the 0-RTT
+encryption keys do not provide full forward secrecy and that the
+the server is not able to guarantee full uniqueness of the handshake
+(non-replayability) without keeping potentially undue amounts of
+state. See {{early-data-indication}} for one mechanism to limit
+the exposure to replay.
+
+The reader should refer to the following references for analysis of the
+TLS handshake {{CHSV16}} {{FGSW16}} {{LXZFH16}}.
+
+## Record Layer {#security-record-layer}
+
+The record layer depends on the handshake producing a strong session
+key which can be used to derive bidirectional traffic keys and nonces.
+Assuming that is true, and the keys are used for no more data than
+indicated in {{limits-on-key-usage}} then the record layer should provide the following
+guarantees:
+
+Confidentiality.
+: An attacker should not be able to determine the plaintext contents
+of a given record. More formally it should not be able to distinguish
+between multiple encryptions of the same versus different plaintexts.
+{::comment}Cite IND-CPA?{:/comment}
+
+Integrity.
+: An attacker should not be able to craft a new record which is
+different from an existing record which will be accepted by the receiver
+with more than negligible probability.
+{::comment}Cite INT-CTXT?{:/comment}
+
+Order protection/non-replayability
+: An attacker should not be able to cause the receiver to accept a
+record which it has already accepted or cause the receiver to accept
+record N+1 without having first processed record N.
+
```

Comment:
I think a TODO is in order for that one :)

On Mon, Jul 11, 2016 at 10:58 AM, Watson Ladd <notifications@github.com>
wrote:

> In draft-ietf-tls-tls13.md
> <https://github.com/tlswg/tls13-spec/pull/525#discussion_r70305891>:
>
> > +: An attacker should not be able to determine the plaintext contents
> > +of a given record. More formally it should not be able to distinguish
> > +between multiple encryptions of the same versus different plaintexts.
> > +{::comment}Cite IND-CPA?{:/comment}
> > +
> > +Integrity.
> > +: An attacker should not be able to craft a new record which is
> > +different from an existing record which will be accepted by the receiver
> > +with more than negligible probability.
> > +{::comment}Cite INT-CTXT?{:/comment}
> > +
> > +Order protection/non-replayability
> > +: An attacker should not be able to cause the receiver to accept a
> > +record which it has already accepted or cause the receiver to accept
> > +record N+1 without having first processed record N.
> > +
>
> Even for DTLS? If we're not folding it in, then this looks good, but if we
> are we need a more complex definition.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tlswg/tls13-spec/pull/525/files/ccf7cf46a7b7a4c3e74d9b041944f94675c9949f#r70305891>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABD1oay7y3wC9lP8GWmirafsWYcWqXQXks5qUoRWgaJpZM4JIstx>
> .
>